test/event-manager.spec.js

```diff
@@ -192,6 +193,8 @@ describe('EventManager', () => {
       const oneCapture = jasmine.createSpy('one-capture');
       const threeCapture = jasmine.createSpy('three-capture');
 
+      spyOn(LogManager.Logger.prototype, 'warn')
```

Comment:
You should do this in a `beforeEach` along with storing the original `warn` method in a local variable and restoring it in `afterEach`, otherwise the spy stays for the remainder of the tests.

Reply:
as per 
https://github.com/jasmine/jasmine/issues/1281#issuecomment-286831790
it will be cleaned automatically
and there is no way to manually restore AFAIK

Reply:
Well, up to you. My suggestion would be this:

```ts
      let originalWarn;
      beforeEach(() => {
        originalWarn = LogManager.Logger.prototype.warn;
        spyOn(LogManager.Logger.prototype, 'warn')
      });
      afterEach(() => {
        LogManager.Logger.prototype.warn = originalWarn;
      });
```

Reply:
done